app/models/goal.py

```diff
@@ -1,5 +1,13 @@
 from app import db
-
+from flask import current_app
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title=db.Column(db.String)
+    tasks = db.relationship("Task", back_populates="goal", lazy=True)
+
+    def to_json_goal(self):
```

Comment:
Minor naming: since this is on the `Goal` class, there's no need to have `goal` repeated in the name, and it causes a bit of likely repetition assuming we name our `Goal` variable `goal` giving us `goal.to_json_goal()`. `to_json` would be fine here, we have the context of it being a `Goal` from the class.